auditing/meilisearch.go

```diff
@@ -400,20 +400,24 @@ func (a *meiliAuditing) migrateIndexSettings(index *meilisearch.Index) error {
 	}
 	diff := &meilisearch.Settings{}
 
-	if current.TypoTolerance != nil && current.TypoTolerance.Enabled != desired.TypoTolerance.Enabled {
-		changesRequired = true
-		diff.TypoTolerance = desired.TypoTolerance
-	}
-
-	if !slices.Equal(current.SortableAttributes, desired.SortableAttributes) {
+	// We'd like to disable the typo tolerance completely, but that's not possible, yet.
+	// TypoTolerance.Enabled is marked as omitempty and therefore prevents overriding true with false.
+	// https://github.com/meilisearch/meilisearch-go/issues/452
+	//
+	// if current.TypoTolerance != nil && current.TypoTolerance.Enabled != desired.TypoTolerance.Enabled {
+	// 	changesRequired = true
+	// 	diff.TypoTolerance = desired.TypoTolerance
+	// }
+
+	if !slicesUnorderedEqual(current.SortableAttributes, desired.SortableAttributes) {
 		changesRequired = true
 		diff.SortableAttributes = desired.SortableAttributes
 	}
-	if !slices.Equal(current.SearchableAttributes, desired.SearchableAttributes) {
+	if !slicesUnorderedEqual(current.SearchableAttributes, desired.SearchableAttributes) {
 		changesRequired = true
 		diff.SearchableAttributes = desired.SearchableAttributes
 	}
-	if !slices.Equal(current.FilterableAttributes, desired.FilterableAttributes) {
+	if !slicesUnorderedEqual(current.FilterableAttributes, desired.FilterableAttributes) {
```

Comment:
Maybe a dump question: Why not simply sort both slices before and use the slices.Equal ?

There is sort.Slice available

Reply:
The updated version does it 👍